modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceSolrIndex.java

```diff
@@ -491,19 +496,27 @@ protected SolrInputDocument createDocument(WorkflowInstance instance) throws Exc
     doc.addField(WORKFLOW_CREATOR_KEY, workflowCreator.getUsername());
     doc.addField(ORG_KEY, instance.getOrganization().getId());
 
-    WorkflowInstance.WorkflowState state = instance.getState();
-    if (!(WorkflowInstance.WorkflowState.SUCCEEDED.equals(state)
-            || WorkflowInstance.WorkflowState.FAILED.equals(state)
-            || WorkflowInstance.WorkflowState.STOPPED.equals(state))) {
+    // Media package used to get the active acl
+    MediaPackage aclMp = mp;
 
-      AccessControlList acl;
-      try {
-        acl = authorizationService.getActiveAcl(mp).getA();
-      } catch (Error e) {
-        logger.error("No security xacml found on media package {}", mp);
-        throw new WorkflowException(e);
+    // If workflow has ended, get the latest acl from the asset manager because there may be no security
+    // attachments in the workspace anymore
+    WorkflowInstance.WorkflowState state = instance.getState();
+    if (state.isTerminated()) {
+      AQueryBuilder query = assetManager.createQuery();
+      AResult result = query.select(query.snapshot())
+              .where(query.mediaPackageId(mp.getIdentifier().compact()).and(query.version().isLatest())).run();
+      if (result.getSize() > 0 && result.getRecords().head().isSome()) {
+        aclMp = result.getRecords().head().get().getSnapshot().get().getMediaPackage();
       }
+    }
+
+    try {
+      AccessControlList acl = authorizationService.getActiveAcl(aclMp).getA();
       addAuthorization(doc, acl);
+    } catch (Error e) {
```

Comment:
Do we need to catch a generic error here?

Reply:
I don't think so (I think this was already there...)